engine/src/main/java/org/terasology/rendering/dag/RenderingGraph.java

```diff
@@ -0,0 +1,140 @@
+    private Config config;
+    private boolean analyzed;
+
+    public RenderingGraph(Context context) {
+        this.context = context;
+        this.nodeMap = Maps.newLinkedHashMap(); // insert order is important for now
+        this.config = context.get(Config.class);
+        this.stateManager = StateManager.createDefault(context);
+        this.analyzed = false;
+    }
+
+    public static RenderingGraph createDefault(Context context) {
+        RenderingGraph pipeline = new RenderingGraph(context);
+        pipeline.add("shadowMap", ShadowMapNode.class);
+        pipeline.add("worldReflective", WorldReflectionNode.class);
+        pipeline.add("backdrop", BackdropNode.class);
+        pipeline.add("skyBands", SkyBandsNode.class);
+        pipeline.add("objectsOpaque", ObjectsOpaqueNode.class);
+        pipeline.add("chunksOpaque", ChunksOpaqueNode.class);
+        pipeline.add("chunksAlphaReject", ChunksAlphaRejectNode.class);
+        pipeline.add("overlays", OverlaysNode.class);
+        pipeline.add("firstPersonView", FirstPersonViewNode.class);
+        pipeline.add("lightGeometry", LightGeometryNode.class);
+        pipeline.add("directionalLights", DirectionalLightsNode.class);
+        pipeline.add("chunksRefractiveReflective", ChunksRefractiveReflectiveNode.class);
+        pipeline.add("outline", OutlineNode.class);
+        pipeline.add("ambientOcclusionPasses", AmbientOcclusionPassesNode.class);
+        pipeline.add("prePostComposite", PrePostCompositeNode.class);
+        pipeline.add("simpleBlendMaterials", SimpleBlendMaterialsNode.class);
+        pipeline.add("lightShafts", LightShaftsNode.class);
+        pipeline.add("initialPostprocessing", InitialPostProcessingNode.class);
+        pipeline.add("downSampleSceneAndUpdateExposure", DownSampleSceneAndUpdateExposureNode.class); // TODO: a rename needed :)
+        pipeline.add("toneMapping", ToneMappedSceneNode.class);
+        pipeline.add("bloomPasses", BloomPassesNode.class);
+        pipeline.add("blurPasses", BlurPassesNode.class);
+        pipeline.add("finalPostProcessing", FinalPostProcessingNode.class);
+        pipeline.analyze();
+        return pipeline;
+    }
+
+    public void analyze() {
+        topologicalOrder = getTopologicalOrder();
+        stateManager.findStateChanges(topologicalOrder);
+        analyzed = true;
+    }
+
+    public void add(AbstractNode node) {
+        nodeMap.put(node.getIdentifier(), node);
+    }
+
+    public <T extends AbstractNode> void add(String identifier, Class<T> type) {
+        // Attempt constructor-based injection first
+        context.put(String.class, identifier);
+        T node = InjectionHelper.createWithConstructorInjection(type, context);
+        // Then fill @In fields
+        InjectionHelper.inject(node, context);
+        node.initialise();
+        this.add(type.cast(node));
+    }
+
+    public Node get(String identifier) {
+        return nodeMap.get(identifier);
+    }
```

Comment:
Let's do it this way: let's keep this functionality, but let's think a little bit more in-depth about what object should store the id. 

Initially I thought all the information related to a node should be stored into a node. But if the purpose of an id is to retrieve a node from the render graph than perhaps the id could just be stored in the map held by the render graph - the node itself would have no id.

If we go along this line I propose the RenderGraph class manipulates nodes -mostly- only by id. Notable exception would be:

`public string add(Node node, string suggestedId)`

This method would return a string because it is possible that the suggested id might be already in use. In this case the returned string would be a slightly modified version of the suggested id. I.e. if a "blur" node already exist and a module wishes to register another "blur" node, the actual id returned could be "blur2".

One potentially trivial issue I'm wondering about is how to name the accessor methods. Should we use methods such as 

add(), get(), remove() or addNode(), getNode(), removeNode()? At this stage, with the render graph only handling nodes, this is perhaps an unnecessary distinction. But as we add edge-related functionality perhaps addNode/getNode/removeNode() would make more sense?

Thoughts?